src/processors/other/LevelDetective.cpp

```diff
@@ -0,0 +1,167 @@
+    levelOutBuffer.setSize (1, numSamples, false, false, true);
+    //if audio input connected, extract level from input signal and assign to levelOutBuffer
+    if (inputsConnected.contains (AudioInput))
+    {
+        //create span to fill audio visualiser buffer
+        nonstd::span<const float> audioChannelData = { buffer.getReadPointer (0), (size_t) numSamples };
+        levelVisualizer.pushChannel (0, audioChannelData);
+        level.setParameters (*attackMsParam, *releaseMsParam);
+        level.processBlock (buffer, levelOutBuffer);
+
+        //create span to fill level visualiser buffer
+        nonstd::span<const float> levelChannelData = { levelOutBuffer.getReadPointer (0), (size_t) numSamples };
+        levelVisualizer.pushChannel (1, levelChannelData);
+    }
+    else
+    {
+        levelOutBuffer.clear();
+    }
+
+    outputBuffers.getReference (LevelOutput) = &levelOutBuffer;
+}
+
+void LevelDetective::processAudioBypassed (AudioBuffer<float>& buffer)
+{
+    const auto numSamples = buffer.getNumSamples();
+    levelOutBuffer.setSize (1, numSamples, false, false, true);
+    if (inputsConnected.contains (AudioInput))
+    {
+        levelOutBuffer.clear();
+        outputBuffers.getReference (LevelOutput) = &levelOutBuffer;
+    }
+}
+
+bool LevelDetective::getCustomComponents (OwnedArray<Component>& customComps, chowdsp::HostContextProvider& hcp)
+{
+    using namespace chowdsp::ParamUtils;
+    class LevelDetectiveEditor : public juce::Component
+    {
+    public:
+        explicit LevelDetectiveEditor (LevelDetectorVisualizer& viz, AudioProcessorValueTreeState& vtState, chowdsp::HostContextProvider& hcp)
+            : visualiser (viz),
+              vts (vtState),
+              attackSlider (*getParameterPointer<chowdsp::FloatParameter*> (vts, attackTag), hcp),
+              releaseSlider (*getParameterPointer<chowdsp::FloatParameter*> (vts, releaseTag), hcp),
+              attackAttach (vts, attackTag, attackSlider),
+              releaseAttach (vts, releaseTag, releaseSlider)
+        {
+            attackLabel.setText ("Attack", juce::dontSendNotification);
+            releaseLabel.setText ("Release", juce::dontSendNotification);
+            attackLabel.setJustificationType (Justification::centred);
+            releaseLabel.setJustificationType (Justification::centred);
+
+            addAndMakeVisible (attackLabel);
+            addAndMakeVisible (releaseLabel);
+
+            for (auto* s : { &attackSlider, &releaseSlider })
+            {
+                s->setSliderStyle (Slider::RotaryHorizontalVerticalDrag);
+                s->setTextBoxStyle (Slider::TextBoxBelow, false, 80, 20);
+                s->setColour (Slider::textBoxHighlightColourId, powerColour.withAlpha (0.55f));
+                s->setColour (Slider::thumbColourId, powerColour);
+                addAndMakeVisible (s);
+            }
+
+            visualiser.backgroundColour = backgroundColour.darker (0.4f);
```

Comment:
So to be able to push this change through, we'll need to update the submodule for chowdsp_utils. So the steps for that are:
- in your local copy create new branch and make a PR with your changes
- once the PR is merged, pull the submodule to the latest commit, and then push the submodule update to this repo

Reply:
Is this different to the branch and PR I created for utils with the changes to waveformView? [https://github.com/Chowdhury-DSP/chowdsp_utils/pull/446]

Reply:
Nope that's the one! Just merged it.